File: typescript/generate-eval-scripts.ts
import * as fs from "node:fs";
import * as path from "node:path";

// Component types to generate scripts for
const componentTypes = ["plugins", "adapters", "wallets"];

// Template for test-eval.ts
const testEvalTemplate = (componentType: string, componentName: string) => `import { runEvals } from "@goat-sdk/core";
import { ChatOpenAI } from "@langchain/openai";
// TODO: Import appropriate wallet client and plugins
import { ALL_TOOLS_DATASET } from "./src/eval";

require("dotenv").config();

// Set up LLM
const llm = new ChatOpenAI({
    modelName: "gpt-4o-mini",
    temperature: 0.3,
});

// TODO: Create wallet client appropriate for this component

async function runTest() {
    try {
        const result = await runEvals(
            ALL_TOOLS_DATASET,
            {
                wallet: null, // TODO: Add appropriate wallet
                plugins: [], // TODO: Add appropriate plugins
                llm,
            },
            "${componentName} Evaluation Tests",
        );

        console.log("\\nEvaluation Summary:");
        console.log(\`Success: \${result.success}\`);
        console.log(\`Total: \${result.results.length}\`);
        console.log(\`Passed: \${result.results.filter((r) => r.passed).length}\`);
        console.log(\`Failed: \${result.results.filter((r) => !r.passed).length}\`);
    } catch (error) {
        console.error("Error running evaluations:", error);
    }
}

runTest();
`;

// Generate scripts for all components
for (const componentType of componentTypes) {
    const componentsPath = path.join(__dirname, "packages", componentType);
    const components = fs
        .readdirSync(componentsPath)
        .filter((dir) => fs.statSync(path.join(componentsPath, dir)).isDirectory());

    for (const component of components) {
        const evalDirPath = path.join(componentsPath, component, "src", "eval");
        const testFilePath = path.join(componentsPath, component, "test-eval.ts");

        // Only generate if eval directory exists but test file doesn't
        if (fs.existsSync(evalDirPath) && !fs.existsSync(testFilePath)) {
            console.log(`Generating test script for ${componentType}/${component}`);
            fs.writeFileSync(testFilePath, testEvalTemplate(componentType, component));
        }
    }
}

console.log("Script generation completed");

File: python/generate_eval_scripts.py
import os
from pathlib import Path

# Component types to generate scripts for
component_types = ["plugins", "adapters", "wallets"]

# Template for test_eval.py
def test_eval_template(component_type, component_name):
    return f'''import os
import asyncio
from dotenv import load_dotenv
from langchain_openai import ChatOpenAI

from goat.evals.utils.run_evals import run_evals
# TODO: Import appropriate wallet client and plugins
from goat_{component_type}.{component_name}.eval import ALL_TOOLS_DATASET

# Load environment variables
load_dotenv()

# TODO: Setup wallet appropriate for this component

# Setup LLM
llm = ChatOpenAI(
    model="gpt-4o-mini",
    temperature=0.3
)

async def run_test():
    try:
        # TODO: Create plugin appropriate for this component
        
        # Run evaluations
        result = await run_evals(
            dataset=ALL_TOOLS_DATASET,
            wallet=None,  # TODO: Add appropriate wallet
            plugins=[],  # TODO: Add appropriate plugins
            llm=llm,
            test_name="{component_name.capitalize()} Evaluation Tests"
        )

        print("\\nEvaluation Summary:")
        print(f"Success: {{result.success}}")
        print(f"Total: {{len(result.results)}}")
        print(f"Passed: {{sum(1 for r in result.results if r.passed)}}")
        print(f"Failed: {{sum(1 for r in result.results if not r.passed)}}")
    except Exception as e:
        print(f"Error running evaluations: {{e}}")

if __name__ == "__main__":
    asyncio.run(run_test())
'''

# Generate scripts for all components
for component_type in component_types:
    components_path = Path("src") / component_type
    components = [d for d in components_path.iterdir() if d.is_dir()]
    
    for component in components:
        eval_dir_path = component / f"goat_{component_type[:-1]}s" / "eval"
        test_file_path = component / "test_eval.py"
        
        # Only generate if eval directory exists but test file doesn't
        if eval_dir_path.exists() and not test_file_path.exists():
            print(f"Generating test script for {component_type}/{component.name}")
            with open(test_file_path, 'w') as f:
                f.write(test_eval_template(component_type[:-1], component.name))

print("Script generation completed")
